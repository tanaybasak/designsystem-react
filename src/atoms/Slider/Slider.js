import React , { useState }from 'react';
import PropTypes from 'prop-types';
import prefix from "../../settings";
import Label from '../Label';

const Slider = ({
  min,
  max,
  step,
  className,
  onChange: onChangeHandler,
  value: propsVal,
  withInputBox,
  ...restProps
}) => {
  const [value, setValue] = useState(propsVal || 0);
  const classnames = `${prefix}-slider ${className}`.trim();

  return (
    <div className={classnames}>
      <Label className={`${prefix}-slider-bottom-range`}>{min}</Label>
      <input
        className={`${prefix}-slider-input`}
        type='range'
        min={min}
        max={max}
        step={step}
        value={value}
        {...restProps}
        onChange={event => {
          setValue(event.currentTarget.value);
          if (onChangeHandler && typeof onChangeHandler === "function") {
            onChangeHandler(event);
          }
        }}
      />
      <Label className={`${prefix}-slider-top-range`}>{max}</Label>
      {withInputBox ? (
        <input
          className={`${prefix}-slider-text-input`}
          type='text'
          value={value}
          onChange={event => {
            setValue(Number(event.currentTarget.value));
          }}
          onBlur={event => {
            let val = Number(event.currentTarget.value);
            val = val > max ? max : val < min ? min : val;
            setValue(val);
          }}
          disabled={restProps.disabled}
        />
      ) : null}
    </div>
  );
}

Slider.propTypes = {
  min: PropTypes.number,
  max: PropTypes.number,
  step: PropTypes.number,
  value: PropTypes.number,
  className: PropTypes.string,
  onChange: PropTypes.func,
  withInputBox: PropTypes.bool
};

Slider.defaultProps = {
  min: 0,
  max: 100,
  step: 1,
  value: 0,
  className: "",
  onChange: () => {},
  withInputBox: true
};

export default Slider;