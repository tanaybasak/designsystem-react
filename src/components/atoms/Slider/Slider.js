import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { prefix } from '../../../settings';

export default function Slider({ min = 0, max = 100, className, ...restProps }) {
    const [value, setValue] = useState(restProps.value || 50);
    const classnames = `${prefix}-slider ${className}`.trim();

    return (
        <div className={classnames}>
            <label className={`${prefix}-slider-bottom-range`}>{min}</label>
            <input className={`${prefix}-slider-input`}
                type="range"
                min={min}
                max={max}
                value={value}
                onChange={event => {
                    setValue(event.currentTarget.value);
                    //restProps.onChange(event);
                }}
            />
            <label className={`${prefix}-slider-top-range`}>{max}</label>
            <input
                className={`${prefix}-slider-text-input`}
                type="text"
                value={value}
                onChange={event => {
                    const val = event.currentTarget.value;
                    const minVal = Number(min);
                    const maxVal = Number(max);
                    if ((Number(val) && Number(val) >= minVal && Number(val) <= maxVal) || !val) {
                        setValue(event.currentTarget.value || 0);
                    }
                }}
            />
        </div>
    );
};

Slider.propTypes = {
    small: PropTypes.bool,
    title: PropTypes.string,
    className: PropTypes.string
};

Slider.defaultProps = {
    small: false,
    title: '',
    className: ''
};