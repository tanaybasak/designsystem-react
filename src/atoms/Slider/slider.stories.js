import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { number, boolean ,text} from '@storybook/addon-knobs';
//@update-path-build-start
import Slider from './Slider';
//@update-path-build-end

storiesOf('Slider', module).add(
  'default',
  () => (
    <Slider
      max={number('Max', 100)}
      min={number('Min', 0)}
      onChange={action('OnChange')}
      step={number('Step', 1)}
      value={50}
      id="basic-slider"
      helperText={text('Helper Text', 'Optional Helper text goes here (max 1000 and min 0)')}
      label={text('Label', 'Basic Slider')}
      title="Slider"
      withInputBox={boolean('With Input Box', true)}
    />
  ),
  {
    info: {
      text: `Description About Slider Component\n
      
      import { Slider } from '@patron/patron-react/slider'
      
      `
    }
  }
);
