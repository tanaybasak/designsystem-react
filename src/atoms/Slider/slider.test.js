import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import prefix from '../../settings';
import Slider from './Slider';

describe('<Slider> component', () => {
    it('snapshots/renders Slider correctly', () => {
        const tree = renderer.create(<Slider />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('simulate change event in Slider and is called Twice', () => {
        const mockCallBack = jest.fn().mockReturnValue(55);
        const wrapper = mount(<Slider onChange={mockCallBack} withInputBox={false} />);
        wrapper.find(`.${prefix}-slider-input`).simulate('change', { currentTarget: { value: 55 } });
        expect(mockCallBack()).toBe(55);
        expect(mockCallBack.mock.calls.length).toBe(2);
        expect(mockCallBack).toHaveBeenCalledTimes(2);
        expect(wrapper.find(`.${prefix}-slider-text-input`).exists()).toBeFalsy();
    });

    it('set props in Slider and renders the same', () => {
        const wrapper = mount(<Slider withInputBox={false} />);
        wrapper.setProps({ value: 34 });
        expect(wrapper.find(`.${prefix}-slider-text-input`).exists()).toBeFalsy();
        expect(wrapper.props().value).toBe(34);
    });

    it('render Slider with inputBox by default', () => {
        const wrapper = mount(<Slider />);
        expect(wrapper.find(`.${prefix}-slider-text-input`).exists()).toBeTruthy();
    });

    it('simulate blur, onchange event in inputbox of Slider', () => {
        const wrapper = shallow(<Slider />);
        wrapper.setProps({value: 56});
        const inputBox = wrapper.find(`.${prefix}-slider-text-input`);
        inputBox.simulate('change', {currentTarget: {value: 5}});
        inputBox.simulate('blur', {currentTarget: {value: 56}});
        expect(wrapper.find(`.${prefix}-slider-text-input`).props().value).toEqual(56);
    });

    it('aceept only Number in inputbox of Slider', () => {
        const wrapper = shallow(<Slider />);
        const inputbox = wrapper.find(`.${prefix}-slider-text-input`);
        inputbox.simulate('change', {
            which:75,
            key: 'k',
            currentTarget: {
                value: 'k'
            }
        });
        expect(wrapper.find(`.${prefix}-slider-text-input`).props().value).not.toBeNaN();
    });
});