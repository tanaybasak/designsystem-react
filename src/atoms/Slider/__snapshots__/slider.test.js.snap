// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider> component snapshots/renders Slider correctly 1`] = `
<div
  className="hcl-slider-wrapper"
>
  <div
    className="hcl-slider"
  >
    <label
      className="hcl-label hcl-slider-bottom-range"
      htmlFor=""
    >
      0
    </label>
    <div
      className="hcl-slider-input-wrapper"
    >
      <input
        className="hcl-slider-input"
        id={null}
        max={100}
        min={0}
        onChange={[Function]}
        onClick={[Function]}
        step={1}
        type="range"
        value={0}
      />
      <div
        className="hcl-range-value"
      >
        <span>
          0
        </span>
      </div>
    </div>
    <label
      className="hcl-label hcl-slider-top-range"
      htmlFor=""
    >
      100
    </label>
    <input
      aria-label={null}
      className="hcl-form-control hcl-slider-text-input"
      data-invalid={false}
      max={100}
      min={0}
      onChange={[Function]}
      step={1}
      type="number"
      value={0}
    />
  </div>
</div>
`;
